presto-spi/src/main/java/com/facebook/presto/spi/type/VarcharType.java

```diff
@@ -29,6 +29,11 @@
     public static final int MAX_LENGTH = Integer.MAX_VALUE;
     public static final VarcharType VARCHAR = new VarcharType(MAX_LENGTH);
 
+    public static VarcharType createUnboundedVarcharType()
```

Comment:
`CONSTANT` can be mistreated and someone can check for the `VARCHAR` type by analogy to `BOOLEAN` like `type.equals(BOOLEAN)`->`type.equals(VARCHAR)`. So, to eliminate the misunderstandings i think it is better to create a method with the `create` word in it.